lxd/migrate.go

```diff
@@ -533,26 +533,53 @@ type MigrationSinkArgs struct {
 	Push      bool
 }
 
-func NewMigrationSink(args *MigrationSinkArgs) (*migrationSink, error) {
+func NewMigrationSink(args *MigrationSinkArgs, push bool) (*migrationSink, error) {
 	sink := migrationSink{
 		src:    migrationFields{container: args.Container},
 		url:    args.Url,
 		dialer: args.Dialer,
+		push:   args.Push,
 	}
 
-	var ok bool
+	if push {
+		sink.allConnected = make(chan bool, 1)
+	}
+
+	var (
+		ok  bool
+		err error
+	)
```

Comment:
Oh, actually, I see why you're doing this now, so it makes sense for it to be defined at this level, but lets just make it "var err error" for consistency :)